Alicerce/Sources/Logging/Formatters/LogItemStringFormatter.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  LogItemStringFormatter.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 03/04/17.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+final class LogItemStringFormatter {
+    
+    static let defaultFormatString = "$DHH:mm:ss.SSS$d $C$L$c $N.$F:$l - $M"
+    let formatString: String
+    let formatter = DateFormatter()
+    let levelColorFormatter: LogItemLevelColorFormatter
+    let levelNameFormatter: LogItemLevelNameFormatter
+        
+    // MARK:- Lifecycle
+    
+    convenience init() {
+        self.init(formatString: LogItemStringFormatter.defaultFormatString,
+                  levelColorFormatter: LogItemLevelColorDefaultFormatter(),
+                  levelNameFormatter: LogItemLevelNameDefaultFormatter())
+    }
+    
+    convenience init(formatString: String) {
+        self.init(formatString: formatString,
+                  levelColorFormatter: LogItemLevelColorDefaultFormatter(),
+                  levelNameFormatter: LogItemLevelNameDefaultFormatter())
+    }
+    
+    convenience init(formatString: String, levelColorFormatter: LogItemLevelColorFormatter) {
+        self.init(formatString: formatString,
+                  levelColorFormatter: levelColorFormatter,
+                  levelNameFormatter: LogItemLevelNameDefaultFormatter())
+    }
+    
+    init(formatString: String, levelColorFormatter: LogItemLevelColorFormatter, levelNameFormatter: LogItemLevelNameFormatter) {
+        self.formatString = formatString
+        self.levelColorFormatter = levelColorFormatter
+        self.levelNameFormatter = levelNameFormatter
+    }
+}
+
+// MARK:- LogItemFormatter
+
+extension LogItemStringFormatter: LogItemFormatter {
+    
+    internal func format(logItem: LogItem) -> String {
+        
+        var text = ""
+        let phrases: [String] = self.formatString.components(separatedBy: "$")
+        
+        for phrase in phrases {
+            if !phrase.isEmpty {
```

Comment:
This condition could be converted into a `guard phrase.isEmpty == false else { continue }`, which would avoid this additional nesting caused by the `if` 😉